pkg/executor/core/executor.go

```diff
@@ -156,3 +160,27 @@ func (ex Executor) verifyReference(ctx context.Context, subjectRef common.Refere
 
 	return types.VerifyResult{IsSuccess: isSuccess, VerifierReports: verifyResults}
 }
+
+func (ex Executor) addNestedVerifierResult(ctx context.Context, referenceDesc ocispecs.ReferenceDescriptor, subjectRef common.Reference, verifyResult *vr.VerifierResult) {
+	verifyParameters := e.VerifyParameters{
```

Comment:
As mentioned by Toddy from the community meeting, is there a risk for infinite looping?  should we create a tracking item to limit the level of nesting allowed, and build more restrictive allowed nested types , e.g. SBOM can have notary signatures as a nested artifact, but we should not have SBOM nested in a SBOM?

Reply:
I don't believe there is a risk of an infinite loop,  but please let me know if you are thinking of different scenario. If you specify you specify nestedReferences as `true` for SBOMs and you have a store that looks like the one below with verifiers for signatures and SBOMs: 
```
image
    |---- SBOM1
          |---- SBOM2
          |---- signature
```
1. Fetch the referrers for the image =>  SBOM1
2. verify the contents of SBOM1
3. Fetch the nested references for SBOM1 => SBOM2 + signature
    a.  verify the signature
    b.  verify the contents of SBOM2
        1.  fetch nested references for SBOM2 => no referrers for this artifact
        2. loop stops and returns no referrers for this artifact
 
Because you have specified that all SBOMs must have nested references, this would cause the nested verification to fail which would set `isSuccess: false`. However, if SBOM2 had a valid signature you should get `isSuccess: true`.

Reply:
thanks for the clarification @joshuaphelpsms. The scenario in mind is more of a malicious attach on ratify. For example could there be  a compromised registry where there are deep level of nested artifacts, in such a case Ratify could be looping and unable to proceed with other requests.  I think this is a existing problem with current code path, just wondering is that a real risk, and if we want to put a limit.

Reply:
Hmm good question. I am trying to think of a scenario where Ratify would be in a infinite loop. It's true that if you had a registry with a deep level of nesting that Ratify would continue down the nested levels as long as it:
1. Has a verifier for that artifact type
2. That artifact type has specified that it has nested references.

It might be warranted to set a limit on the level of nesting for performance reasons, but I can't think of a reason why it would be in an infinite loop. Open to hearing other thoughts on this or other scenarios that I am not thinking about

Reply:
ah you are right, there are no infinite loop, the loops will go as deep as the number of nested artifacts exist on the image.